packages/wallet-adapter-core/src/WalletCore.ts

```diff
@@ -172,51 +172,58 @@ export class WalletCore extends EventEmitter<WalletCoreEvents> {
     }
   }
 
-  /** 
-  Connects a wallet to the app. If a wallet is already connected,
-  we first disconnect the current connected wallet and then connect the selected wallet.
-  On connect success, we set the current account and the network, and keeping the selected wallet
-  name in LocalStorage to support autoConnect function.
-
-  @param walletName. The wallet name we want to connect as a WalletName type.
-  @emit emits "connect" event
-  @throws WalletConnectionError
-  */
+  /**
+   * We first make sure we can connect a dapp to a wallet.
+   * If all good, we connect the wallet by calling `this.connectWallet`
+   * @param walletName. The wallet name we want to connect.
+   */
   async connect(walletName: WalletName): Promise<void | string> {
-    try {
-      const selectedWallet = this._wallets?.find(
-        (wallet: Wallet) => wallet.name === walletName
-      );
-
-      if (!selectedWallet) return;
-
-      if (this._connected) {
-        // if the selected wallet is already connected, we don't need to connect again
-        if (this.wallet?.name === walletName)
-          throw new WalletConnectionError(
-            `${walletName} wallet is already connected`
-          ).message;
+    const selectedWallet = this._wallets?.find(
+      (wallet: Wallet) => wallet.name === walletName
+    );
+
+    if (!selectedWallet) return;
+
+    if (this._connected) {
+      // if the selected wallet is already connected, we don't need to connect again
+      if (this.wallet?.name === walletName)
```

Comment:
Wanted to make sure, Is there any chance that wallet is connected (this._connected) but this.wallet? return `null` or `None`? or shall we include the check `if (this._connected && this.wallet)`?

Reply:
yep good catch, should probably do `if (selectedWallet === walletName)`

Reply:
done